packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -65,37 +68,65 @@ export class EditorPage {
 	 * @param {Page} page The underlying page.
 	 * @param param0 Keyed object parameter.
 	 * @param param0.target Target editor type. Defaults to 'simple'.
-	 */
-	constructor( page: Page, { target = 'simple' }: { target?: SiteType } = {} ) {
+	 * @param param0.blockTheme Whether block-based theme is used. Defaults to 'false'.
+	 */
+	constructor(
+		page: Page,
+		{ target = 'simple', blockTheme = false }: { target?: SiteType; blockTheme?: boolean } = {}
+	) {
+		// The first step is to determine whether the test site is running a
+		// Gutenframe, otherwise known as a Calypsofy iframe.
+		// Typically, a Gutenframe is found on Simple sites and encapsulates the
+		// entire editor window.
+		// Atomic sites typically do not feature a Gutenframe and thus the editor
+		// window is exposed in the DOM.
+		// For both Simple and Atomic, the relevant `body` root element is used when resolving
+		// the Locator. This is to present a unified behavior when other methods reference the
+		// `editorWindow`.
 		if ( target === 'atomic' ) {
-			// For Atomic editors, there is no iFrame - the editor is
-			// part of the page DOM and is thus accessible directly.
-			this.editor = page.locator( selectors.editor );
+			this.editorWindow = page.locator( selectors.editor );
+		} else {
+			this.editorWindow = page.frameLocator( selectors.editorFrame ).locator( selectors.editor );
+		}
+
+		// The second step is to locate the iframe that exists within the
+		// editor canvas as of Gutenberg 14.9.1 when using newer block-based themes.
+		// If the parameter `blockTheme` is true, the editor canvas is hidden inside a new
+		// iframe and must be pierced to be visible.
+		if ( blockTheme ) {
+			this.editorCanvas = this.editorWindow
+				.frameLocator( selectors.editorCanvasFrame )
+				.locator( 'body' );
 		} else {
-			// For Simple editors, the editor is located within an iFrame
-			// and thus it must first be extracted.
-			this.editor = page.frameLocator( selectors.editorFrame ).locator( selectors.editor );
+			this.editorCanvas = this.editorWindow;
```

Comment:
This is the core of the PR.

I explored several ways for `EditorPage` to perform this task without any intervention from the user, not even with parameters. Such attempts were not successful because the constructor is synchronous and thus we do not have insight into whether the editor features an inner iframe or not.

Alternatives I considered were to force the spec writer to call a series of initialization methods, but I decided against that because of the issues it would cause when the spec writers forget to call such actions.

```
const editorPage = new EditorPage();
// What if the spec writer forgets the next line?
// Or what if `EditorPage`changes? It would require changing this across all specs.
await editorPage.initialize();
```

One path I had not explored yet is to have a private async method that is called when anything needs to interact with the editor canvas. The method can silently look for an inner iframe and update the class property if it finds the frame.